core/src/main/java/apoc/trigger/TriggerHandlerRead.java

```diff
@@ -0,0 +1,228 @@
+                params.putAll((Map<String, Object>) data.get("params"));
+            }
+            Map<String, Object> selector = (Map<String, Object>) data.get("selector");
+            if ((!(boolean)data.get("paused")) && when(selector, phase)) {
+                try {
+                    params.put("trigger", name);
+                    Result result = tx.execute((String) data.get("statement"), params);
+                    Iterators.count(result);
+                } catch (Exception e) {
+                    log.warn("Error executing trigger " + name + " in phase " + phase, e);
+                    exceptions.put(name, e.getMessage());
+                }
+            }
+        });
+        if (!exceptions.isEmpty()) {
+            throw new RuntimeException("Error executing triggers "+ exceptions);
+        }
+    }
+
+    private boolean when(Map<String, Object> selector, Phase phase) {
+        if (selector == null) return phase == Phase.before;
+        return Phase.valueOf(selector.getOrDefault("phase", "before").toString()) == phase;
+    }
+
+    @Override
+    public void start() throws Exception {
+        updateCache();
+        long refreshInterval = apocConfig().getInt(TRIGGER_RELOAD, TRIGGER_RELOAD_DEFAULT_VALUE);
+        restoreTriggerHandler = jobScheduler.scheduleRecurring(Group.STORAGE_MAINTENANCE, () -> {
+            if (getLastUpdate() > lastUpdate) {
+                updateCache();
+            }
+        }, refreshInterval, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private long getLastUpdate() {
+        return withSystemDb(tx -> {
+            Node node = tx.findNode(SystemLabels.ApocTriggerMeta, SystemPropertyKeys.database.name(), db.databaseName());
+            return node == null ? 0L : (long) node.getProperty(SystemPropertyKeys.lastUpdated.name());
+        });
+    }
+    
+    private void updateCache() {
+        activeTriggers.clear();
+
+        lastUpdate = System.currentTimeMillis();
+
+        withSystemDb(tx -> {
+            tx.findNodes(SystemLabels.ApocTrigger,
+                    SystemPropertyKeys.database.name(), db.databaseName()).forEachRemaining(
+                    node -> activeTriggers.put(
+                            (String) node.getProperty(SystemPropertyKeys.name.name()),
+                            MapUtil.map(
+                                    "statement", node.getProperty(SystemPropertyKeys.statement.name()),
+                                    "selector", Util.fromJson((String) node.getProperty(SystemPropertyKeys.selector.name()), Map.class),
+                                    "params", Util.fromJson((String) node.getProperty(SystemPropertyKeys.params.name()), Map.class),
+                                    "paused", node.getProperty(SystemPropertyKeys.paused.name())
+                            )
+                    )
+            );
+            return null;
+        });
+
+        reconcileKernelRegistration();
+    }
```

Comment:
I struggle to imagine of a use case where a user would want a trigger that could only read data. It would seem utterly pointless to me. 

I can imagine use cases where users might want to trigger some side effect, such as writing data to another database - which is definitely possible with Extended. 

I'm just wondering if it would be a sensible restriction to only triggers to be executed on WRITE transactions to the leader of a database. I think this would be too much of a breaking change though.

However, do we know what happens when a follower of a database finally catches up with the leader? Does that happen in a transaction that would re-execute the trigger?

Reply:
Actually, it might be worth checking whether TransactionEventListener is even invoked for READ transactions. This [answer](https://neo4j.slack.com/archives/C02JJGEPD/p1666802698640259) from Sergey to a different question has me thinking that might not be the case 

If only LEADERS are allowed to execute transactions because they are the only ones that can handle writes, then our code should reflect that, and we shouldn't be registering TransactionEventListeners on DBMS members which don't have write access to the database for the given trigger.

Reply:
Had a call with Nacho. He made a great point that we should be registering the event listeners even on followers, so as to avoid any missed transactions in case the follower gets promoted to leader.